index.src.html

```diff
@@ -3772,6 +3780,26 @@
 {{ImageBitmapOptions}} {{ImageBitmapOptions/colorSpaceConversion}}. Setting this
 value to "none" disables color space conversion.
 
+### WebGPU ### {#videoframe-webgpu}
+
+WebCodecs extends [[WebGPU]] to allow importing {{VideoFrame}} objects via
+{{GPUDevice/importExternalTexture()}}, producing opaque {{GPUExternalTexture}}
+objects, which can be read from WebGPU shaders.
+The definition of the {{GPUExternalTextureDescriptor/source}} member of
+{{GPUExternalTextureDescriptor}} is modified to allow {{VideoFrame}}:
+
+<xmp>
+partial dictionary GPUExternalTextureDescriptor {
+    required (HTMLVideoElement or VideoFrame) source;
+};
+</xmp>
+
+The lifetime of such a {{GPUExternalTexture}} object is implicitly tied to the
+lifetime of the source {{VideoFrame}} object.
```

Comment:
The WebGPU spec currently says that `GPUExternalTexture` objects are destroyed after a turn of the microtask queue, in which case having them keep the underlying resource alive should be fine.

Is it expected that this approach will be changed? Are the sampler configurations heavy enough that it would be useful to retain them for longer? If they are retained longer, is their own lifetime insufficient for some reason (eg. being GC-based)?

Reply:
My proposal is to have GPUExternalTexture tied to different implicit lifetimes depending on how it's created. If from a VideoFrame, it uses the lifetime of the VideoFrame. If from an HTMLVideoElement, I'm proposing a change to use the lifetime of the currently presented frame, rather than an implicit microtask: https://github.com/gpuweb/gpuweb/pull/2302
(The microtask thing was our initial "most cautious" approach but we're finding that it's not the most practical solution for a number of reasons.)

Reply:
Thanks. What sort of synchronization do you expect apps to need to do to correctly complete rendering before they can call `VideoFrame.close()`?

Reply:
Synchronization in WebGPU is implicit. Apps just need to submit() any work that uses the external texture before close()ing. So an implementation might need to flush commands on close(), or hold the VideoFrame(Handle) alive internally until flush, and use some internal synchronization to make sure the different subsystems are synchronized.

I'm pretty sure this wouldn't be much different from using VideoFrame as a texImage2D() source in WebGL, when boiled down.

Reply:
Text you settled on (mentioning transfer and close()) works for me. I'm also happy to support adding steps to "Close Video Frame" if you'd prefer. IIUC, this would be something like: for each texture referencing this frame, set textures [[detached]] to true.